site/en/tutorials/estimators/cnn.ipynb

```diff
@@ -0,0 +1,1014 @@
+        "cross-entropy of the logits layer and our labels. Let's configure our model to\n",
+        "optimize this loss value during training. We'll use a learning rate of 0.001 and\n",
+        "[stochastic gradient descent](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)\n",
+        "as the optimization algorithm:\n",
+        "\n",
+        "```\n",
+        "if mode == tf.estimator.ModeKeys.TRAIN:\n",
+        "  optimizer = tf.train.GradientDescentOptimizer(learning_rate=0.001)\n",
+        "  train_op = optimizer.minimize(\n",
+        "      loss=loss,\n",
+        "      global_step=tf.train.get_global_step())\n",
+        "  return tf.estimator.EstimatorSpec(mode=mode, loss=loss, train_op=train_op)\n",
+        "```"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "rEJPnXAzS6m9",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Note: For a more in-depth look at configuring training ops for Estimator model functions, see [\"Defining the training op for the model\"](../../guide/custom_estimators.md#defining-the-training-op-for-the-model) in the [\"Creating Estimations in tf.estimator\"](../../guide/custom_estimators.md) tutorial."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "QQuGDWvHTAib",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Add evaluation metrics\n",
+        "\n",
+        "To add accuracy metric in our model, we define `eval_metric_ops` dict in EVAL\n",
+        "mode as follows:\n",
+        "\n",
+        "```\n",
+        "eval_metric_ops = {\n",
+        "    \"accuracy\": tf.metrics.accuracy(\n",
+        "        labels=labels, predictions=predictions[\"classes\"])\n",
+        "}\n",
+        "return tf.estimator.EstimatorSpec(\n",
+        "    mode=mode, loss=loss, eval_metric_ops=eval_metric_ops)\n",
+        "```"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "Y2Bwe-AdTRzX",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "<a id=\"train_eval_mnist\"></a>\n",
+        "## Training and Evaluating the CNN MNIST Classifier\n",
+        "\n",
+        "We've coded our MNIST CNN model function; now we're ready to train and evaluate\n",
+        "it."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "6EC9aOY2TTLU",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Load Training and Test Data\n",
+        "\n",
+        "First, let's load our training and test data with the following code:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "ccobb0qETV-S",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "# Load training and eval data\n",
+        "mnist = tf.contrib.learn.datasets.load_dataset(\"mnist\")\n",
```

Comment:
```
# Load training and eval data
((train_data, train_labels),
 (eval_data, eval_labels)) = tf.keras.datasets.mnist.load_data()

train_data = (train_data/255.0).astype(np.float32)
train_labels = train_labels.astype(np.int32)

eval_data = (eval_data/255.0).astype(np.float32)
eval_labels = eval_labels.astype(np.int32)
```

Gets us numpy arrays, without contrib.
That way there's no work to do rewording the text.

Reply:
Done, thanks

Reply:
I think we should convert to float32 first and then divide. The numpy array is float64 which is a costlier division than float32. Not that it matters here but its a good practice.

Reply:
Update to Mark's snippet ...
@yashk2810 Can you add a patch to this PR or paste a snippet for your suggestion?

Reply:
# Load training and eval data
((train_data, train_labels),
 (eval_data, eval_labels)) = tf.keras.datasets.mnist.load_data()

train_data = train_data/np.float32(255)

# not required
train_labels = train_labels.astype(np.int32)

eval_data = eval_data/np.float32(255)

# not required
eval_labels = eval_labels.astype(np.int32)